### DISCUSSION TITLE
Floe Riing Config

### DISCUSSION THREAD

**Discussion by kevinhaycock** [devcompl/TTController#100] [Configuration] [answered]
Hello,

Firstly I would like to say a BIG THANK YOU to the creator of this excellent TTController!!!

Now on to my issue, I have been reading through the docs and attempted to make my config how I would like it.

**My configuration;**
===========
```
Controllers
-------------------------------
Name: Riing Plus
VendorId: 9802
ProductId: 8102
Ports:
       
        Id: 1
        Data: [PortId: 1, Speed: 45, Rpm: 0, Unknown: 255]
        Identifier: [9802, 8102, 1]

        Id: 2
        Data: [PortId: 2, Speed: 45, Rpm: 980, Unknown: 255]
        Identifier: [9802, 8102, 2]

        Id: 3
        Data: [PortId: 3, Speed: 45, Rpm: 971, Unknown: 255]
        Identifier: [9802, 8102, 3]

        Id: 4
        Data: [PortId: 4, Speed: 45, Rpm: 977, Unknown: 255]
        Identifier: [9802, 8102, 4]

        Id: 5
        Data: [PortId: 5, Speed: 35, Rpm: 0, Unknown: 255]
        Identifier: [9802, 8102, 5]

Available effect types:
Flow_Extreme, Flow_Fast, Flow_Normal, Flow_Slow, Spectrum_Extreme, Spectrum_Fast, Spectrum_Normal, Spectrum_Slow, Ripple_Extreme, Ripple_Fast, Ripple_Normal, Ripple_Slow, Blink_Extreme, Blink_Fast, Blink_Normal, Blink_Slow, Pulse_Extreme, Pulse_Fast, Pulse_Normal, Pulse_Slow, Wave_Extreme, Wave_Fast, Wave_Normal, Wave_Slow, ByLed, Full

Plugins
-------------------------------
Valid plugins:
        TTController.Plugin.BlinkEffect.dll
        TTController.Plugin.ByLedColorEffect.dll
        TTController.Plugin.FlowEffect.dll
        TTController.Plugin.FullColorEffect.dll
        TTController.Plugin.PulseEffect.dll
        TTController.Plugin.RawEffect.dll
        TTController.Plugin.RazerConnectEffect.dll
        TTController.Plugin.RippleEffect.dll
        TTController.Plugin.SensorEffect.dll
        TTController.Plugin.SnakeEffect.dll
        TTController.Plugin.SoundEffect.dll
        TTController.Plugin.SpectrumEffect.dll
        TTController.Plugin.WaveEffect.dll
        TTController.Plugin.CopySpeedController.dll
        TTController.Plugin.PwmSpeedController.dll
        TTController.Plugin.StaticSpeedController.dll
        TTController.Plugin.AlwaysTrigger.dll
        TTController.Plugin.LogicTrigger.dll
        TTController.Plugin.OneTimeTrigger.dll
        TTController.Plugin.ProcessTrigger.dll
        TTController.Plugin.PulseTrigger.dll
        TTController.Plugin.SensorTrigger.dll
        TTController.Plugin.DpsgController.dll
        TTController.Plugin.RiingController.dll
        TTController.Plugin.RiingPlusController.dll
        TTController.Plugin.RiingTrioController.dll

Sensors
-------------------------------
Intel Core i7-2700K:
        CPU Core #1 (/intelcpu/0/temperature/0): 43
        CPU Core #2 (/intelcpu/0/temperature/1): 37
        CPU Core #3 (/intelcpu/0/temperature/2): 41
        CPU Core #4 (/intelcpu/0/temperature/3): 42
        CPU Package (/intelcpu/0/temperature/4): 43
NVIDIA GeForce GTX 760:
        GPU Core (/nvidiagpu/0/temperature/0): 28
NVIDIA GeForce GTX 760:
        GPU Core (/nvidiagpu/1/temperature/0): 32
NVIDIA GeForce GTX 760:
        GPU Core (/nvidiagpu/2/temperature/0): 30
Samsung SSD 860 EVO 1TB:
        Temperature (/hdd/0/temperature/0): 27
-------------------------------
Press any key to continue...
```

**Now what I am attempting todo;**
======================

1. Have my pump pulse slowly red and max flow ( I do not know if the rate of flow is controllable)

2. Have the top fans slowly pulse blue

So I managed to get ColorGradient to work on the top fans, but no luck with the slow pulse.

I have been trying to get this to work for a while as, the ThermalTake software is junk, however I am now stuck.

Currently I have attempted to get the "pump" plusing red.

Please be nice, I have no coding expirence.

**Here is the config I managed to throw together;**
=============================
```json
{
  "Profiles": [
    {
      "Name": "TopFans",
      "Ports": [
        [9802, 8102, 2],
        [9802, 8102, 3],
        [9802, 8102, 4]
      ],
      "SpeedControllers": [
        {
          "Type": "PwmSpeedController", 
          "Config": {
            "CurvePoints": [
              [20, 30],
              [30, 50],
              [50, 60],
              [55, 75],
              [65, 100]
            ],
            "Sensors": [
              "/intelcpu/0/temperature/4",
              "/intelcpu/0/temperature/3",
              "/intelcpu/0/temperature/2",
              "/intelcpu/0/temperature/1",
              "/intelcpu/0/temperature/0"
            ],
            "SensorMixFunction": "Average",
            "MinimumChange": 4,
            "MaximumChange": 8,
            "Trigger": {
              "Type": "AlwaysTrigger"
            }
          }
        }
      ],
      "Effects": [
        {
          "Type": "SensorEffect",
          "Config": {
            "Sensors": ["/intelcpu/0/temperature/4"],
            "ColorGradient": [
              [40, [0, 0, 64]],
              [50, [64, 64, 64]],
              [62, [64, 0, 0]]
            ],
            "Trigger": {
              "Type": "AlwaysTrigger"
            }
          }
        }
      ]
    },
    {
      "Name": "Pump",
      "Ports": [
        [9802, 8102, 1]
      ],
      "SpeedControllers": [],
      "Effects": [
        {
          "Config": {
            "BrightnessStep": 0.33,
            "Color": {
              "Full": [255, 0, 0]
            },
            "Trigger": {
              "Type": "OneTimeTrigger"
            }
          }
        }
      ]
    },
  ],
  "ComputerStateProfiles": [
    {
      "StateType": "Shutdown",
      "Ports": [
        [9802, 8102, 1],
        [9802, 8102, 2],
        [9802, 8102, 3],
        [9802, 8102, 4]
      ],
      "Speed": 45,
      "Color": {
        "Full": [64, 0, 0]
      }
    },
    {
      "StateType": "Boot",
      "Ports": [
        [9802, 8102, 1],
        [9802, 8102, 2],
        [9802, 8102, 3],
        [9802, 8102, 4]
      ],
      "Speed": 45,
      "Color": {
        "Full": [0, 0, 0]
      }
    }
  ],
  "PortConfigs": [
    {
      "Ports": [
        [9802, 8102, 2],
        [9802, 8102, 3],
        [9802, 8102, 4]
      ],
      "Config": {
        "Name": "Top Fan",
        "DeviceType": "FloeRiing",
      }
    }
  ],
  "SensorConfigs": [
    {
      "Sensors": ["/intelcpu/0/temperature/4"],
      "Config": {
        "CriticalValue": 68
      }
    }
  ]
}
```

Forgot to mention, I have no idea what port 5 is for???

Any help is greatly appreciated

**Comment by MoshiMoshi0** [comment, marked as answer]
> Forgot to mention, I have no idea what port 5 is for???

The service spits out all ports even if you have nothing connected there, there is no way to tell how many and which ports are used.

Btw, which version are you using because the debug report you posted looks like from an old version.

Try this:
```json
{
  "Profiles": [
    {
      "Name": "TopFans",
      "Ports": [
        [9802, 8102, 2],
        [9802, 8102, 3],
        [9802, 8102, 4]
      ],
      "SpeedControllers": [
        {
          "Type": "PwmSpeedController", 
          "Config": {
            "CurvePoints": [
              [20, 30],
              [30, 50],
              [50, 60],
              [55, 75],
              [65, 100]
            ],
            "Sensors": [
              "/intelcpu/0/temperature/4",
              "/intelcpu/0/temperature/3",
              "/intelcpu/0/temperature/2",
              "/intelcpu/0/temperature/1",
              "/intelcpu/0/temperature/0"
            ],
            "SensorMixFunction": "Average",
            "MinimumChange": 4,
            "MaximumChange": 8,
            "Trigger": {
              "Type": "AlwaysTrigger"
            }
          }
        }
      ],
      "Effects": [
        {
          "Type": "PulseEffect",
          "Config": {
            "BrightnessStep": 0.05,
            "Color": {
              "Full": [0, 0, 255]
            },
            "Trigger": {
              "Type": "AlwaysTrigger"
            }
          }
        }
      ]
    },
    {
      "Name": "Pump",
      "Ports": [
        [9802, 8102, 1]
      ],
      "SpeedControllers": [],
      "Effects": [
        {
          "Type": "PulseEffect",
          "Config": {
            "BrightnessStep": 0.05,
            "Color": {
              "Full": [255, 0, 0]
            },
            "Trigger": {
              "Type": "AlwaysTrigger"
            }
          }
        }
      ]
    },
  ],
  "ComputerStateProfiles": [
    {
      "StateType": "Shutdown",
      "Ports": [
        [9802, 8102, 1],
        [9802, 8102, 2],
        [9802, 8102, 3],
        [9802, 8102, 4]
      ],
      "Speed": 45,
      "Color": {
        "Full": [64, 0, 0]
      }
    },
    {
      "StateType": "Boot",
      "Ports": [
        [9802, 8102, 1],
        [9802, 8102, 2],
        [9802, 8102, 3],
        [9802, 8102, 4]
      ],
      "Speed": 45,
      "Color": {
        "Full": [0, 0, 0]
      }
    }
  ],
  "PortConfigs": [
    {
      "Ports": [
        [9802, 8102, 1],
      ],
      "Config": {
        "DeviceType": "FloeRiing",
      }
    }
  ],
  "SensorConfigs": [
    {
      "Sensors": ["/intelcpu/0/temperature/4"],
      "Config": {
        "CriticalValue": 68
      }
    }
  ]
}
```

**Comment by kevinhaycock** [comment]
> Btw, which version are you using because the debug report you posted looks like from an old version.

Nice catch, I was using TTController_0.7.1.0-alpha as I had trouble getting it to work.

I have just updated to the latest version and applied the config you posted and it works!!!!

Thank you so much for your help and rapid reply!!!!!

**Comment by kevinhaycock** [comment]
I don't want to be cheeky and attempt use this as my personal request forum, however I would just like to ask is there an effect similar to radar effect?

[https://64.media.tumblr.com/19a8878d54686bcefebb34db2eca404e/tumblr_otk6ex3PLO1rn5gv3o1_500.gifv](url)

For some reason clicking on the link does not work but if it's pasted into a browser, no issue.

The pulse effect is now working perfectly.

I would like something similar to a radar effect on the pump.

Anyone who has a simliar setup here is the current config.


{
  "Profiles": [
    {
      "Name": "TopFans",
      "Ports": [
        [9802, 8102, 2],
        [9802, 8102, 3],
        [9802, 8102, 4]
      ],
      "SpeedControllers": [
        {
          "Type": "PwmSpeedController", 
          "Config": {
            "CurvePoints": [
              [20, 30],
              [30, 50],
              [50, 60],
              [55, 75],
              [65, 100]
            ],
            "Sensors": [
              "/intelcpu/0/temperature/4",
              "/intelcpu/0/temperature/3",
              "/intelcpu/0/temperature/2",
              "/intelcpu/0/temperature/1",
              "/intelcpu/0/temperature/0"
            ],
            "SensorMixFunction": "Average",
            "MinimumChange": 4,
            "MaximumChange": 8,
            "Trigger": {
              "Type": "AlwaysTrigger"
            }
          }
        }
      ],
      "Effects": [
        {
          "Type": "PulseEffect",
          "Config": {
            "BrightnessStep": 0.02,
            "Color": {
              "Full": [0, 0, 50]
            },
            "Trigger": {
              "Type": "AlwaysTrigger"
            }
          }
        }
      ]
    },
    {
      "Name": "Pump",
      "Ports": [
        [9802, 8102, 1]
      ],
      "SpeedControllers": [],
      "Effects": [
        {
          "Type": "PulseEffect",
          "Config": {
            "BrightnessStep": 0.05,
            "Color": {
              "Full": [255, 0, 0]
            },
            "Trigger": {
              "Type": "AlwaysTrigger"
            }
          }
        }
      ]
    },
  ],
  "ComputerStateProfiles": [
    {
      "StateType": "Shutdown",
      "Ports": [
        [9802, 8102, 1],
        [9802, 8102, 2],
        [9802, 8102, 3],
        [9802, 8102, 4]
      ],
      "Speed": 45,
      "Color": {
        "Full": [64, 0, 0]
      }
    },
    {
      "StateType": "Boot",
      "Ports": [
        [9802, 8102, 1],
        [9802, 8102, 2],
        [9802, 8102, 3],
        [9802, 8102, 4]
      ],
      "Speed": 45,
      "Color": {
        "Full": [0, 0, 0]
      }
    }
  ],
  "PortConfigs": [
    {
      "Ports": [
        [9802, 8102, 1],
      ],
      "Config": {
        "DeviceType": "FloeRiing",
      }
    }
  ],
  "SensorConfigs": [
    {
      "Sensors": ["/intelcpu/0/temperature/4"],
      "Config": {
        "CriticalValue": 68
      }
    }
  ]
}

**Comment by MoshiMoshi0** [comment]
You mean like this?:

```json
{
  "Type": "RippleEffect",
  "Config": {
    "Length": 5,
    "TickInterval": 3,
    "RippleColor": {
      "Gradient": [
        [1, [0,0,0]],
        [0.5, [0,64,0]],
        [0, [0, 255, 0]]
      ]
    },
    "BackgroundColor": {
      "Full": [0, 0, 0]
    },
    "Trigger": {
      "Type": "AlwaysTrigger"
    }
  }
}
```

**Comment by kevinhaycock** [comment]
Excellent Sir!

Merry Christmas to you and your family!!!!

Cannot thank you enough.

**Comment by kevinhaycock** [comment]
Final config for anyone whos wants their FloeRiing setup as the follwoing;

1. Pump RippleEffect (Radar), Green (Thanks Moshi, I got converted from Red)
2. Radiator fans (TopFans) pulsing slowly Blue

This setup is for a I7-2700K @ 5.2Ghz  max temp around 70c considered safe now days, but it's easy to change.

**Configuration;**
===========

{
  "Profiles": [
    {
      "Name": "TopFans",
      "Ports": [
        [9802, 8102, 2],
        [9802, 8102, 3],
        [9802, 8102, 4]
      ],
      "SpeedControllers": [
        {
          "Type": "PwmSpeedController", 
          "Config": {
            "CurvePoints": [
              [20, 30],
              [30, 50],
              [50, 60],
              [55, 75],
              [65, 100]
            ],
            "Sensors": [
              "/intelcpu/0/temperature/4",
              "/intelcpu/0/temperature/3",
              "/intelcpu/0/temperature/2",
              "/intelcpu/0/temperature/1",
              "/intelcpu/0/temperature/0"
            ],
            "SensorMixFunction": "Average",
            "MinimumChange": 4,
            "MaximumChange": 8,
            "Trigger": {
              "Type": "AlwaysTrigger"
            }
          }
        }
      ],
      "Effects": [
        {
          "Type": "PulseEffect",
          "Config": {
            "BrightnessStep": 0.015,
            "Color": {
              "Full": [0, 0, 40]
            },
            "Trigger": {
              "Type": "AlwaysTrigger"
            }
          }
        }
      ]
    },
    {
      "Name": "Pump",
      "Ports": [
        [9802, 8102, 1]
      ],
      "SpeedControllers": [],
      "Effects": [
        {
  "Type": "RippleEffect",
  "Config": {
    "Length": 5,
    "TickInterval": 3,
    "RippleColor": {
      "Gradient": [
        [1, [0,0,0]],
        [0.5, [0,48,0]],
        [0, [0, 128, 0]]
      ]
    },
    "BackgroundColor": {
      "Full": [0, 0, 0]
    },
    "Trigger": {
      "Type": "AlwaysTrigger"
    }
  }
}
      ]
    },
  ],
  "ComputerStateProfiles": [
    {
      "StateType": "Shutdown",
      "Ports": [
        [9802, 8102, 1],
        [9802, 8102, 2],
        [9802, 8102, 3],
        [9802, 8102, 4]
      ],
      "Speed": 45,
      "Color": {
        "Full": [64, 0, 0]
      }
    },
    {
      "StateType": "Boot",
      "Ports": [
        [9802, 8102, 1],
        [9802, 8102, 2],
        [9802, 8102, 3],
        [9802, 8102, 4]
      ],
      "Speed": 45,
      "Color": {
        "Full": [0, 0, 0]
      }
    }
  ],
  "PortConfigs": [
    {
      "Ports": [
        [9802, 8102, 1],
      ],
      "Config": {
        "DeviceType": "FloeRiing",
      }
    }
  ],
  "SensorConfigs": [
    {
      "Sensors": ["/intelcpu/0/temperature/4"],
      "Config": {
        "CriticalValue": 68
      }
    }
  ]
}

=======================
Just a quick note, I have managed to learn powershell to a level that I can build a gui for this........

If you would like to work together to make a GUI where it's "clickable" let me know, it would be things like effects entered to your txt config through ps script.

All the best

K

**Comment by kevinhaycock** [comment]
MOoshi moshi

Rules!!!

**Comment by kevinhaycock** [comment]
OK, so I feel the need to do this now, so I will start to build the PS script to do this (a clickable GUI) give me a week.

Including GIFs showing the effects.

It will happen!

I will post the reply here.

**Reply by MoshiMoshi0** [reply]
Nice! Cool to see other people make stuff for this project.
If you do make it I think it would be better to make a new general thread where you will be able to post updates and stuff, just to keep it in one dedicated thread.

**Reply by kevinhaycock** [reply]
Will do. Alreadly started! Screens coming!!!

Just a quick question, a general thread where?

I'm new to this github stuff.

Many thanks

K

**Comment by kevinhaycock** [comment]
[https://ibb.co/NKFSNJj](url)

We have begun

**Comment by kevinhaycock** [comment]
I will have to look into create a new general thred, however here is a screen of the work so far (by no means complete)

https://ibb.co/Rb8hxWZ

I will stop posting in this thread now.
